tests/baselines/reference/aliasInstantiationExpressionGenericIntersectionNoCrash.errors.txt

```diff
@@ -0,0 +1,23 @@
+aliasInstantiationExpressionGenericIntersectionNoCrash.ts(10,1): error TS2352: Conversion of type 'typeof Err<U> & typeof Err<U>' to type 'typeof Err<U> & typeof Err<U>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
+  Type 'typeof Err<U> & typeof Err<U>' is not comparable to type 'typeof Err<U>'.
+    Type 'ErrImpl<number>' is not comparable to type 'ErrImpl<string>'.
+      Type 'number' is not comparable to type 'string'.
+
+
+==== aliasInstantiationExpressionGenericIntersectionNoCrash.ts (1 errors) ====
+    class ErrImpl<E> {
+      e!: E;
+    }
+    
+    declare const Err: typeof ErrImpl & (<T>() => T);
+    
+    type ErrAlias<U> = typeof Err<U>;
+    
+    declare const e: ErrAlias<number>;
+    e as ErrAlias<string>;
+    ~~~~~~~~~~~~~~~~~~~~~
+!!! error TS2352: Conversion of type 'typeof Err<U> & typeof Err<U>' to type 'typeof Err<U> & typeof Err<U>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.

```

Comment:
what the hell is this error message? 😕

Reply:
Yes, exactly. I don't get it. I have debugged for hours trying to figure out _why_ these types when printed don't have their type parameters mapped. The mapper is there, the types are compared, queried, etc, correctly, but when you actually look at them, the type vars don't get replaced!

Reply:
it's also weird to me that we have `'typeof Err<U> & typeof Err<U>' in the first place... how is that not just `typeof Err<U>`?

Reply:
welcome to hell!

Reply:
Consider this a plea of "please someone else be nerd sniped by this too"

Reply:
I think it's a bug in the printer implementation for instantiation expression types in `createAnonymousTypeNode` - namely, it goes back to the declared type when deciding what to print for an instantiation expression on a type query node:
```ts
                    const isInstantiationExpressionType = !!(getObjectFlags(type) & ObjectFlags.InstantiationExpressionType);
                    if (isInstantiationExpressionType) {
                        const instantiationExpressionType = type as InstantiationExpressionType;
                        if (isTypeQueryNode(instantiationExpressionType.node)) {
                            const typeNode = serializeExistingTypeNode(context, instantiationExpressionType.node);
                            if (typeNode) {
                                return typeNode;
                            }
                        }
                        if (context.visitedTypes?.has(typeId)) {
                            return createElidedInformationPlaceholder(context);
                        }
                        return visitAndTransformType(type, createTypeNodeFromObjectType);
                    }
```
usually we use `serializeTypeForDeclaration` and not `serializeExistingTypeNode` directly because it checks that the existing node is actually the same type as the one we're looking at.

Reply:
Aha, so dropping that `typeof` special case and letting it walk shows more info:

```
aliasInstantiationExpressionGenericIntersectionNoCrash.ts(10,1): error TS2352: Conversion of type '{ new (): ErrImpl<number>; prototype: ErrImpl<any>; } & (() => number)' to type '{ new (): ErrImpl<string>; prototype: ErrImpl<any>; } & (() => string)' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type '{ new (): ErrImpl<number>; prototype: ErrImpl<any>; } & (() => number)' is not comparable to type '{ new (): ErrImpl<string>; prototype: ErrImpl<any>; }'.
    Type 'ErrImpl<number>' is not comparable to type 'ErrImpl<string>'.
      Type 'number' is not comparable to type 'string'.
```

It's checking the two intersections against each other, so it goes down into one branch.

Reply:
Looking at the changes here, I think that removing this special case is the right thing to do, honestly. The way instantiation expressions seem to work is different than type references; the type arguments that it was instantiated with just aren't preserved at all.

It makes me feel like instantiation expressions should instead be some form of TypeReference; all of the places that check InstantiationExpression appear to also look for Reference too...